go/vt/vtgate/planbuilder/testdata/dml_cases.json

```diff
@@ -4889,5 +4889,49 @@
     "comment": "Unsupported update statement with a replica target destination",
     "query": "update `user[-]@replica`.user_metadata set id=2",
     "plan": "VT09002: update statement with a replica target"
+  },
+  {
+    "comment": "insert row values smaller than number of columns",
+    "query": "insert into unsharded(one, two, three, four) values (1, 2, 3)",
+    "plan": {
+      "QueryType": "INSERT",
+      "Original": "insert into unsharded(one, two, three, four) values (1, 2, 3)",
+      "Instructions": {
+        "OperatorType": "Insert",
+        "Variant": "Unsharded",
+        "Keyspace": {
+          "Name": "main",
+          "Sharded": false
+        },
+        "TargetTabletType": "PRIMARY",
+        "Query": "insert into unsharded(one, two, three, four) values (1, 2, 3)",
+        "TableName": "unsharded"
+      },
+      "TablesUsed": [
+        "main.unsharded"
+      ]
+    }
+  },
+  {
+    "comment": "insert row values greater than number of columns",
+    "query": "insert into unsharded(one, two, three) values (1, 2, 3, 4)",
+    "plan": {
+      "QueryType": "INSERT",
+      "Original": "insert into unsharded(one, two, three) values (1, 2, 3, 4)",
+      "Instructions": {
+        "OperatorType": "Insert",
+        "Variant": "Unsharded",
+        "Keyspace": {
+          "Name": "main",
+          "Sharded": false
+        },
+        "TargetTabletType": "PRIMARY",
+        "Query": "insert into unsharded(one, two, three) values (1, 2, 3, 4)",
+        "TableName": "unsharded"
+      },
+      "TablesUsed": [
+        "main.unsharded"
+      ]
+    }
```

Comment:
How do these tests verify that your code changes work? Your code change is meant to fail the planning of a query with incorrect column and row counts, but this plan test just means that planning doesn't fail...

Reply:
Then should I write a e2e test somewhere?

Reply:
No no, I tested it out today, all you need to do is change the table in the insert query. For the `unsharded` keyspaces, we have code that bypasses almost the entire planning, that is why the queries don't fail. Just use a table in sharded keyspace like `user`.

So change your test queries to `insert into user(one, two, three, four) values (1, 2, 3)` and `insert into user(one, two, three) values (1, 2, 3, 4)`

Reply:
I have even verified that the first query used to fail with the array index out of bounds error prior to your changes. So just changing these tests would be enough to verify your changes work.